### ISSUE TITLE
Array multiply-add not compiling

### ISSUE THREAD

**Issue by jeanlaroche** [serge-sans-paille/pythran#870]
This code no longer compiles in MacOS. It does compile in linux.
```
iimport numpy as np

#pythran export test(float[][],float[][],int,int,float[][],int,int)
def test(sig2,sigOut,N2,NOut,rampDownCrossFade,M1,M2):
    LCf = len(rampDownCrossFade)
    sigOut[NOut:NOut +LCf, ] += sig2[M2:M2+LCf,] * (1-rampDownCrossFade)  # <- fails.
    #sigOut[NOut:NOut + LCf, ] += sig2[M2:M2 + LCf, ]  # <- compiles
    return NOut + N2
```

The error is:

> In file included from /var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmpfsHEtE.cpp:9:
In file included from /Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/numpy_texpr.hpp:4:
In file included from /Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/ndarray.hpp:39:
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/numpy_gexpr.hpp:205:47: error: invalid operands to binary expression ('long' and 'const typename tuple_element<0UL, tuple<contiguous_slice> >::type' (aka 'const __type_pack_element<0UL, (anonymous namespace)::pythonic::types::contiguous_slice>'))
              std::make_tuple(std::get<0>(t0) * std::get<0>(t1)),
                              ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/numpy_gexpr.hpp:255:27: note: in instantiation of template class '(anonymous namespace)::pythonic::types::details::merge_gexpr<std::__1::tuple<const long>, std::__1::tuple<(anonymous namespace)::pythonic::types::contiguous_slice> >' requested here
                          merge_gexpr<std::tuple<S...>, std::tuple<Sp...>>{}(
                          ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/numpy_gexpr.hpp:262:19: note: in instantiation of template class '(anonymous namespace)::pythonic::types::details::make_gexpr<const (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<double, 2> &>, const long> &, (anonymous namespace)::pythonic::types::contiguous_slice>' requested here
      -> decltype(details::make_gexpr<Arg, S...>{}(std::forward<Arg>(arg),
                  ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/numpy_gexpr.hpp:544:21: note: while substituting deduced template arguments into function template 'make_gexpr' [with Arg = const (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<double, 2> &>, const long> &, S = <(anonymous namespace)::pythonic::types::contiguous_slice>]
        -> decltype(make_gexpr(*this, s0));
                    ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/operator_/sub.hpp:15:50: note: in instantiation of template class '(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<double, 2> &>, const long>' requested here
  auto sub(A const &a, B const &b) -> decltype(a - b)
                                                 ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/operator_/sub.hpp:22:18: note: while substituting deduced template arguments into function template 'sub' [with A = double, B = (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<double, 2> &>, const long>]
  DEFINE_FUNCTOR(pythonic::operator_, sub);
                 ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/include/types/numpy_expr.hpp:476:33: note: (skipping 11 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
    using value_type = decltype(Op()(std::declval<
                                ^
/Users/jlaroche/temp/JeanPythran/pythran/pythonic/types/ndarray.hpp:367:5: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::ndarray<double, 2>::initialize_from_expr<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::mul, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<double, 2> &, (anonymous namespace)::pythonic::types::contiguous_slice>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::sub, (anonymous namespace)::pythonic::types::broadcast<double, long>, (anonymous namespace)::pythonic::types::numpy_texpr<(anonymous namespace)::pythonic::types::ndarray<double, 2> > &> > >' requested here
    initialize_from_expr(expr);

**Comment by serge-sans-paille** [comment]
I confirm it works fine on Linux :-/

**Comment by serge-sans-paille** [comment]
@jeanlaroche what's the output when using branch ``pythran/fix/remove-universal-reference`` ?

**Comment by jeanlaroche** [comment]
The assert you added fails:

> pythran/pythonic/include/types/numpy_gexpr.hpp:354:5: error: static_assert failed "no modifiers on slices"
    static_assert(
    ^
/Users/jlaroche/temp/pythran/pythran/pythonic/operator_/mul.hpp:15:48: note: in instantiation of template class '(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::ndarray<double, 2>, (anonymous namespace)::pythonic::types::slice, const long>' requested here
  auto mul(A const &a, B const &b) -> decltype(a *b)
                                               ^
/Users/jlaroche/temp/pythran/pythran/pythonic/operator_/mul.hpp:22:18: note: while substituting deduced template arguments into function template 'mul' [with A = (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::ndarray<double, 2>, (anonymous namespace)::pythonic::types::slice, const long>, B = (anonymous namespace)::pythonic::types::broadcast<double, double>]
  DEFINE_FUNCTOR(pythonic::operator_, mul);

**Comment by serge-sans-paille** [comment]
Can you tracj the origin of this const long then?

**Comment by jeanlaroche** [comment]
BTW I'm using this simplified version of the bug:
```
import numpy as np

#pythran export test(float[][],float[][])
def test(sig2,rampDownCrossFade):
    A = (sig2 * 2) * rampDownCrossFade  # <- fails.
    return A
```

**Comment by serge-sans-paille** [comment]
Fixed by #879 , as confirmed by @jeanlaroche on IRC #pythran
